UnityProject/Assets/Scripts/Player/PlayerOnlySyncValues.cs

```diff
@@ -37,13 +88,38 @@ public class PlayerOnlySyncValues : NetworkBehaviour
 		//NightVision
 		private CameraEffectControlScript cameraEffectControlScript;
 
+		private LightingSystem lightingSystem;
+
 		#endregion
 
 		#region LifeCycle
 
+		public void ClientOnPlayerLeaveBody()
+		{
+			ApplyValues(true);
+		}
+
+		public void ClientOnPlayerTransferProcess()
+		{
+			ApplyValues(false);
+		}
+
+		public void ApplyValues(bool Leaving) //TODO Add your custom synchronised value here
+		{
+			OverrideLocalPlayer = true;
+			//SyncXrayState(No need to apply values, if (Leaving) ?  = Reset the default InitialXRay : Otherwise apply current state XRay);
+			SyncXrayState(true, Leaving ? XRay.initialState : XRay);
+			TRaySyncMode(TRayScanner.Mode.Wires, Leaving ? TRayDefaultcurrentMode : TRayCurrentMode);
+			var nn = new NightVisionGoggles.NightVisionData(b: true) { isOn = true};
```

Comment:
```suggestion
			var nn = new NightVisionGoggles.NightVisionData(b: true);
```

Already using the constructor, is broken but needs to be fixed in other comment

Reply:
b: true doesn't link to anything